docs/developer-guide/working-with-rules-new.md

```diff
@@ -0,0 +1,613 @@
+# Working with Rules
+
+Each ESLint rule has two files: a source file in the `lib/rules` directory and a test file in the `tests/lib/rules` directory. Both files should be named with the rule ID (i.e., `no-eval.js` for rule ID `no-eval`) The basic source code format for a rule is:
+
+```js
+/**
+ * @fileoverview Rule to flag use of an empty block statement
+ * @author Nicholas C. Zakas
+ * @copyright 2014 Nicholas C. Zakas. All rights reserved.
+ * See LICENSE in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+
+    },
+    create: function(context) {
+
+        return {
+            // properties go here
+        };
+    }
+};
+
+module.exports.schema = [
+    // JSON Schema for rule options goes here
+];
+```
+
+**Important:** Rule submissions will not be accepted unless they are in this format.
+
+**Note:** Limitations listed above only apply when submitting a new rule to core ESLint repository. If you are working on a custom rule or plugin, naming conventions and availability of unit tests are up to you.
+
+## Rule Basics
+
+Each rule have to export a single object with several required properties.
+
+```js
+module.exports = {
+    meta: {
+        docs: {
+            description: "Short rule description",
+            category: "Best Practices",
+            recommended: false
+        },
+        schema: []
+    },
+    create: function(context) {
+        return {
+            "Identifier": function(node) {
+                // do something with node
+        }
+    }
+}
+```
+
+Meta property contains information about rule's metadata, such as documentation and schema. `docs` property can contain the following keys:
+
+* description - Short description of the rule as it shows in the rule's list
+* category - Category that this rule falls under (see [rules list](http://eslint.org/docs/rules/) for details)
+* recommended - Whether this rule is included in `eslint:recommended` configuration
+* fixable - String explaining what type of fix this rule supports. Can be `whitespace` or `functional`. Omit this property if the rule is not fixable
```

Comment:
`fixable` should be outside of the `docs` object because it should affect more than just documentation.